### PR TITLE
units get promotions and xp bonuses from CS buildings

### PR DESCRIPTION
**Pull request by SimonCeder**
As described in #4602.
Newly gifted units from militaristic city states start with bonus XP and promotions as if they were built by the CS.

### CODE REVIEW

**Comment by logicminimal** [comment]
@ravignir if a city state has two cities then would only the cspitsl count?

**Comment by SimonCeder** [comment]
Yes, this uses the buildings from the city state's capital

File: core/src/com/unciv/models/ruleset/unit/BaseUnit.kt [hunk 1]
```diff
@@ -334,47 +334,51 @@ class BaseUnit : INamed, INonPerpetualConstruction, ICivilopediaText {
...
+                // Deprecated since 3.15.9
                 cityConstructions.cityInfo.getMatchingUniques("New [] units start with [] Experience") +
                 cityConstructions.cityInfo.getLocalMatchingUniques("New [] units start with [] Experience in this city")
-            //
+        //
         ) {
             if (unit.matchesFilter(unique.params[0]))
                 XP += unique.params[1].toInt()
         }
         unit.promotions.XP = XP
```
Comment (SomeTroglodyte) [inline core/src/com/unciv/models/ruleset/unit/BaseUnit.kt:357]: Sounds like an earlier `Military Units gifted from City-States start with [] XP` gets overwritten here

Comment (SimonCeder) [reply core/src/com/unciv/models/ruleset/unit/BaseUnit.kt:357]: It absolutely does

**Pull request merged**